src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -828,18 +784,14 @@ public StringBuilder Append(StringBuilder value, int startIndex, int count)
                 ThrowHelper.ThrowStartIndexArgumentOutOfRange_ArgumentOutOfRange_Index();
             }
 
-            if (count > 0)
-            {
-                if (value == this)
-                    Append(value.ToString(startIndex, count));             
-                else
-                    AppendCore(value, startIndex, count);
-            }
-            return this;
+            return AppendCore(value, startIndex, count);
         }
 
-        private void AppendCore(StringBuilder value, int startIndex, int count)
+        private StringBuilder AppendCore(StringBuilder value, int startIndex, int count)
         {
+            if (count > 0 && value == this)
```

Comment:
Nit: The `value == this` condition should be first because of it is the one more likely to be false.

Reply:
Why is that important? Per Andy it doesn't matter for codegen?
https://github.com/dotnet/coreclr/pull/15786#discussion_r161060827

Reply:
If you have a condition like `if (rarely_false && rarely_true)`, it is always better for perf to write it as `if (rarely_true && rarely_false)` - assuming that both conditions are cheap and side-effect free.

It does not make a huge difference ... just a nit.

Reply:
Ah of course, I misunderstood I thought you meant swapping the if and else clauses.